service/worker/indexer/esProcessor.go

```diff
@@ -162,6 +156,7 @@ func (p *ESProcessorImpl) bulkAfterAction(id int64, requests []bulk.GenericBulka
 						tag.WorkflowRunID(rid),
 						tag.WorkflowDomainID(domainID))
 					p.ackKafkaMsg(key)
+					continue
```

Comment:
This will miss failure metrics `p.scope.IncCounter(metrics.ESProcessorFailures)`

Reply:
These might not should be considered as failures, the version conflicts I found are caused by the disordered messages from Kafka, which mean we have short workflows, sometimes for example the closed messages published earlier than the upsert workflow execution calls, we will have version conflicts and we ack to prevent retry by purpose.

Reply:
We used the taskID as the visibility message version, which can make sure the older messages won't override the newer ones. That is why we got these conflicts